### ISSUE TITLE
[Bug] firebase Realtime database task sometimes never returns when path not found

### ISSUE THREAD

**Issue by shlighter** [firebase/quickstart-unity#1062]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version:2021.1.3f1
  * Firebase Unity SDK version:7.1.0
  * Source you installed the SDK: unitypackage (.unitypackage or Unity Package Manager)
  * Problematic Firebase Component: realtime database(Auth, Database, etc.)
  * Other Firebase Components in use: Auth, realtime database, functions,storage(Auth, Database, etc.)
  * Additional SDKs you are using: AdMob(Facebook, AdMob, etc.)
  * Platform you are using the Unity editor on: windows (Mac, Windows, or Linux)
  * Platform you are targeting: android (iOS, Android, and/or desktop)
  * Scripting Runtime: IL2CPP(Mono, and/or IL2CPP)

#### Relevant Code:
```
`          Debug.Log("before");
            var task = await FirebaseDatabase.DefaultInstance.GetReference(path).GetValueAsync().ContinueWith(t => t);
            if (task.IsFaulted || task.IsCanceled) { Debug.Log("fail"); }
            Debug.Log("after");

           //Output:  "before"`
```
It's very strange behavior, I run the command and nothing happens when the path doesn't exist, the task never returns or fails, I also checked the rules which are fine by the rules simulator. And tried Try+catch, but nothing...

It only happens when the path doesn't exist
#### Steps to reproduce:

Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?
What's the issue repro rate? (eg 100%, 1/5 etc)

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

**Comment by paulinon** [comment]
Hi @shlighter,

Typically, the task result is being read in a function after the lambda operator (=>) as seen in this [example](https://firebase.google.com/docs/database/unity/retrieve-data#read_data_once). With that, could you try this code and see if it makes any difference:


```
FirebaseDatabase.DefaultInstance.GetReference(path).GetValueAsync().ContinueWith(task => {
        if (task.IsFaulted || task.IsCanceled) { 
            Debug.Log("fail"); 
        }
        Debug.Log("after");
});
```

Thanks!

**Comment by shlighter** [comment]
Hi @paulinon  thanks for your answer, I tried your approach and got the same issue, there is no task result,  I don't get the "fail" or "after" logs.  I want to mention that it only happens on Windows Desktop Unity Editor.  It works fine when building to Android or ios.  and it only happens when the path doesn't exist

**Comment by paulinon** [comment]
Hi @shlighter,

I noticed that you're using an old version of the SDK. Could you try using version 7.2.0 of the SDK so that we can confirm if this issue persists?

**Comment by shlighter** [comment]
The new SDK didn’t solve the issue

**Comment by paulinon** [comment]
Hi @shlighter,

Since you mentioned that this issue only happens in the editor, could you also try using an [LTS](https://unity3d.com/unity/qa/lts-releases) version of the Unity editor and see if it persists?

**Comment by shlighter** [comment]
It doesn't work still

**Comment by shlighter** [comment]
You can try it yourself,
Unity 2021.1.3f1 Windows unity editor-  try to access a task which path that doesn't exist, you'll notice it never returns

**Comment by DellaBitta** [comment]
Hi @shlighter,

I attempted to reproduce this on Unity 2021.1.12f1 with the code that you provided. However `await` doesn't work since `GetValueAsync` isn't tagged with `async`.  As @paulinon mentioned the recommended way to wait for `GetValueAsync` is to use `ContinueWith`, or even better, `ContinueWithOnMainTask`

Unfrotuantely I wasn't able to reproduce the problem in the editor on Windows or Mac even with these methods.  Here's the following code:

```
void QueryNonexistantPath() {
      string path = "ThisDoesntExist";
      Debug.Log("before");
      FirebaseDatabase.DefaultInstance.GetReference(path).GetValueAsync().ContinueWith(task => {
        Debug.Log("QueryNonexistantPath task complete");
        if (task.IsFaulted || task.IsCanceled) { 
          Debug.Log("fail");
          return;
        }
        Debug.Log("Result: " + task.Result);
      } );
    }
```

I altered the Database Quickstart to invoke `QueryNonexistantPath()` on a button press. The continuation is invoked as I see `QueryNonexistantPath task complete` in my logs, and the Result is logged as expected:  `Result: DataSnapshot { key = ThisDoesntExist, value =  }`.

If this code doesn't work for you then I recommend starting with a fresh quickstart project and testing it there, and then adding any/all of your other dependencies to see at which point it starts failing.  Let me know how it goes!
